### DISCUSSION TITLE
Finding the Weight of IfcProducts

### DISCUSSION THREAD

**Discussion by RMurphySST** [IfcOpenShell/IfcOpenShell#2829] [Q&A]
Hi,
I currently have an IFC file with a few Beams and Columns in it. Is there anyway to get the weight of these beams through the IfcOpenShell library in Python?

**Comment by Moult** [comment]
If the weight is already calculated, you can do ifcopenshell.util.element.get_psets(element) to get all of its properties and quantities (some sneaky vendors put quantities in their properties instead).

**Comment by aothms** [comment]
Otherwise if you have their density and no other information you can calculate their volume from the evaluated geometries. `ifcopenshell.geom.create_shape()` and `OCC.BRepGProp.BRepGProp_VolumeProperties` with pythonocc or https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/ifcopenshell/util/shape.py#L28 with mesh.
